src/vip/batch_address_test_tool/queue.clj

```diff
@@ -1,82 +1,98 @@
-      (when (nil? @rabbit-connection)
-        (if (< attempt max-retries)
-          (do (Thread/sleep (* attempt 1000))
-              (recur (inc attempt)))
-          (do (log/error "Connecting to RabbitMQ failed. Bailing.")
-            (throw (ex-info "Connecting to RabbitMQ failed" {:attempts attempt})))))))
-  (reset! rabbit-channel
-          (let [ch (lch/open @rabbit-connection)]
-            (le/topic ch (config [:rabbitmq :exchange]) {:durable false :auto-delete true})
-            (log/info "RabbitMQ topic set.")
-            (setup-consumer ch handler-fn)
-            ch)))
+(defn stop-consumer [consumer-id]
+  (squishy/stop-consumer consumer-id))
 
-(defn publish-to-queue
-  "Sends the payload to the given queue, printing it as a EDN string"
-  [payload queue-name]
-  (log/debug (pr-str payload) "->" queue-name)
-  (lb/publish @rabbit-channel
-              ""
-              queue-name
-              (pr-str payload)
-              {:content-type "application/edn"}))
+;; Much/most of this is copied from `data-processor` verbatim, and if we had
+;; a better config story here, we could make it into a library of some kind
+
+(defn sns-client
+  [access-key secret-key region]
+   (aws/client
+    {:api                  :sns
+     :region               region
+     :credentials-provider (credentials/basic-credentials-provider
+                            {:access-key-id     access-key
+                             :secret-access-key secret-key})}))
+
+(def sns-client-memo (memoize sns-client))
+
+(defn default-sns-client
+  []
+   ;; memoize the default sns-client invocation so that we
+   ;; can avoid making extra thread pools over time.
+   ;; see https://github.com/cognitect-labs/aws-api/issues/80
+  (sns-client-memo (config [:aws :creds :access-key])
+                   (config [:aws :creds :secret-key])
+                   (config [:aws :region])))
+
+(defn- publish
+  [sns-client topic payload]
+  (aws/invoke sns-client {:op :Publish
+                          :request {:TopicArn topic
+                                    :Message payload}}))
+
+(defn publish-success
+  "Publish a successful feed processing message to the topic."
+  ([payload]
+   (publish-success (default-sns-client)
+                    (config [:aws :sns :address-test-success-arn])
+                    payload))
+  ([sns-client topic payload]
+   (let [json-payload (json/write-str payload)]
+     (log/info "publishing success to " topic " with payload " json-payload)
+     (publish sns-client topic json-payload))))
+
+(defn publish-failure
+  "Publish a failed feed processing message to the topic."
+  ([payload]
+   (publish-failure (default-sns-client)
+                    (config [:aws :sns :address-failure-failure-arn])
+                    payload))
+  ([sns-client topic payload]
+   (let [json-payload (json/write-str payload)]
+     (log/info "publishing failure to " topic " with payload " json-payload)
```

Comment:
You could use the format string version: `(log/infof "publishing failure to %s with payload %s" topic json-payload)`

Reply:
I did not know about `log/infof`, thanks!